### PR TITLE
Redis connectionTask error handling improvements

### PR DESCRIPTION
**Pull request by JoeHegarty**
This change improves logging and error handling around the Redis connectionTask.

### CODE REVIEW

File: src/main/java/cloud/orbit/actors/cluster/impl/RedisOrbitClient.java [hunk 1]
```diff
@@ -30,30 +30,32 @@
...
-    private Timer connectionTimer;
+    private final Timer connectionTimer;
 
     public RedisOrbitClient(final RedissonClient redisClient, final Integer messagingHealthcheckInterval)
     {
-        this.messagingHealthcheckInterval = messagingHealthcheckInterval;
         this.redisClient = redisClient;
         this.isConnected = redisClient.getNodesGroup().pingAll();
         this.connectionTimer = new Timer();
+
```
Comment (ShawnTuatara) [inline src/main/java/cloud/orbit/actors/cluster/impl/RedisOrbitClient.java]: nit: remove lines that weren't changed (when you were putting the lock in). Shows up a few times in the PR and is noise on the actual changes.

File: src/main/java/cloud/orbit/actors/cluster/impl/RedisOrbitClient.java [hunk 2]
```diff
@@ -68,40 +70,63 @@ public void run()
...
-                    break;
+                    try
+                    {
+                        redisClient.getTopic(subscription.getLeft()).removeAllListeners();
+                        redisClient.getTopic(subscription.getLeft()).addListener(subscription.getRight());
+
+                    }
+                    catch (Exception e)
+                    {
+                        logger.warn("Could not resubscribe, this may be transient", e);
```
Comment (ShawnTuatara) [inline src/main/java/cloud/orbit/actors/cluster/impl/RedisOrbitClient.java]: Include context of what subscription it was trying to add a listener too?

File: src/main/java/cloud/orbit/actors/cluster/impl/RedisOrbitClient.java (hunk 1 above)
Comment (JoeHegarty) [reply src/main/java/cloud/orbit/actors/cluster/impl/RedisOrbitClient.java]: Should be cleaned up now.

File: src/main/java/cloud/orbit/actors/cluster/impl/RedisOrbitClient.java [hunk 3]
```diff
@@ -68,40 +70,63 @@ public void run()
...
+                        subscribedAll = false;
+                        break;
+                    }
                 }
+
+                isConnected = subscribedAll;
             }
+        }
+        catch(Throwable e)
```
Comment (ShawnTuatara) [inline src/main/java/cloud/orbit/actors/cluster/impl/RedisOrbitClient.java]: nit: formatting space

**Review by ShawnTuatara — CHANGES_REQUESTED** [review]

File: src/main/java/cloud/orbit/actors/cluster/impl/RedisOrbitClient.java (hunk 2 above)
Comment (JoeHegarty) [reply src/main/java/cloud/orbit/actors/cluster/impl/RedisOrbitClient.java]: Context added.

File: src/main/java/cloud/orbit/actors/cluster/impl/RedisOrbitClient.java (hunk 3 above)
Comment (JoeHegarty) [reply src/main/java/cloud/orbit/actors/cluster/impl/RedisOrbitClient.java]: Fixed

File: src/main/java/cloud/orbit/actors/cluster/impl/RedisOrbitClient.java [hunk 4]
```diff
@@ -77,31 +78,49 @@ public void subscribe(final String channelId, final MessageListener<Object> mess
...
-                    subscribedAll = false;
-                    break;
+                    try
+                    {
+                        redisClient.getTopic(subscription.getLeft()).removeAllListeners();
+                        redisClient.getTopic(subscription.getLeft()).addListener(subscription.getRight());
+                    }
+                    catch (Exception e)
+                    {
+                        logger.warn("Could not resubscribe to '{}'\n{}", subscription.getLeft(), e);
```
Comment (ShawnTuatara) [inline src/main/java/cloud/orbit/actors/cluster/impl/RedisOrbitClient.java]: Don't need to `\n` the exception. SLF4J interprets the last parameter properly if it is `Throwable`.

https://www.slf4j.org/faq.html#paramException

**Review by ShawnTuatara — CHANGES_REQUESTED** [review]

File: src/main/java/cloud/orbit/actors/cluster/impl/RedisOrbitClient.java (hunk 4 above)
Comment (JoeHegarty) [reply src/main/java/cloud/orbit/actors/cluster/impl/RedisOrbitClient.java]: Really? I never knew that, that's kinda nifty. I will fix.

**Review by ShawnTuatara — APPROVED** [review]

**Comment by JoeHegarty** [comment]
Thanks @ShawnTuatara

**Pull request merged**